vertical-pod-autoscaler/pkg/recommender/input/cluster_feeder.go

```diff
@@ -192,13 +202,60 @@ type clusterStateFeeder struct {
 	memorySaveMode      bool
 	controllerFetcher   controllerfetcher.ControllerFetcher
 	recommenderName     string
+	historySource       HistorySource
+	promHistoryConfig   history.PrometheusHistoryProviderConfig
+}
+
+func (feeder *clusterStateFeeder) Init() error {
+	switch feeder.historySource {
+	case Checkpoints:
+		klog.Infof("Using checkpoints as a history provider")
+		feeder.initFromCheckpoints()
+	case Prometheus:
+		klog.Infof("Using prometheus as a history provider")
+
+		provider, promInitErr := history.NewPrometheusHistoryProvider(feeder.promHistoryConfig)
+		if promInitErr != nil {
+			klog.Errorf("Could not initialize history provider")
+			return promInitErr
+		}
+
+		historyInitErr := feeder.initFromHistoryProvider(provider)
+		if historyInitErr != nil {
+			klog.Errorf("Failed to load prometheus history")
+			return historyInitErr
+		}
+	case None:
+		klog.Infof("Running without a history provider")
+	default:
+		klog.Errorf("Wrong history provider option")
+		return fmt.Errorf("history provider option is not set. Supported values: prometheus, none, checkpoint")
+	}
+	return nil
 }
 
-func (feeder *clusterStateFeeder) InitFromHistoryProvider(historyProvider history.HistoryProvider) {
+func GetHistorySourceFromArg(historySource string) (HistorySource, error) {
+	switch historySource {
+	case "checkpoint":
+		return Checkpoints, nil
+	case "prometheus":
+		return Prometheus, nil
+	case "none":
+		return None, nil
+	default:
+		return Undefined, fmt.Errorf("storage option '%s' is not supported. Supported values: prometheus, none, checkpoint", historySource)
+	}
+}
+
+func (feeder *clusterStateFeeder) initFromHistoryProvider(historyProvider history.HistoryProvider) (historyInitError error) {
+	historyInitError = nil
```

Comment:
I think this is nil by default. Would you mind making the return unnamed and do `var historyInitError error` here instead?